File: ad_library/src/main/java/com/adgrowth/internal/helpers/ScreenHelpers.java
package com.adgrowth.internal.helpers;

import android.app.Activity;
import android.content.Context;
import android.content.pm.ActivityInfo;
import android.content.res.Configuration;

import com.adgrowth.adserver.enums.AdOrientation;

public class ScreenHelpers {
    public static AdOrientation getOrientation(Context context) {
        int orientation = context.getResources().getConfiguration().orientation;

        return orientation == Configuration.ORIENTATION_LANDSCAPE ? AdOrientation.LANDSCAPE : AdOrientation.PORTRAIT;
    }

    public static void setOrientation(Activity context, AdOrientation orientation) {
        context.setRequestedOrientation(orientation == AdOrientation.LANDSCAPE ? ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE : ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);

    }
}

File: ad_library/src/main/kotlin/com/adgrowth/internal/integrations/adserver/helpers/ScreenHelpers.kt
package com.adgrowth.internal.integrations.adserver.helpers


import android.app.Activity
import android.content.pm.ActivityInfo
import android.content.res.Configuration
import android.content.res.Resources
import com.adgrowth.adserver.enums.AdOrientation


object ScreenHelpers {
    @JvmStatic
    fun getOrientation(): AdOrientation {
        val orientation = Resources.getSystem().configuration.orientation
        return if (orientation == Configuration.ORIENTATION_LANDSCAPE) AdOrientation.LANDSCAPE else AdOrientation.PORTRAIT
    }

    fun setOrientation(context: Activity, orientation: AdOrientation) {
        when (orientation) {
            AdOrientation.PORTRAIT -> context.requestedOrientation =
                ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
            AdOrientation.LANDSCAPE -> context.requestedOrientation =
                ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
        }
    }
}
